### PR TITLE
feat: implement Microsoft Entra ID authentication via Firebase

### PR DESCRIPTION
**Pull request by treaks0**
- Add MicrosoftAuth.kt with OAuth provider integration for Microsoft Entra ID
- Add Microsoft OAuth intent-filter in AndroidManifest.xml for authentication flow
- Enable Google Services plugin and Firebase Auth dependencies in build.gradle.kts
- Support tenant validation and JWT token extraction from Microsoft tokens
- Configure OAuth scopes: openid, email, profile, User.Read for complete user access
- Add comprehensive test suite with 25+ test cases covering all authentication scenarios
- Consolidate all Microsoft Auth tests into single MicrosoftAuthComprehensiveTest.kt file
- Achieve meaningful test coverage for authentication logic and edge cases
- Format code with ktfmt for consistent code style

### CODE REVIEW

File: app/src/main/java/com/android/sample/auth/MicrosoftAuth.kt [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+/**
+ * Microsoft OAuth authentication using Firebase Auth. Provides tenant validation and error
+ * handling.
+ */
+object MicrosoftAuth {
+  private const val TAG = "MicrosoftAuth"
+
+  // Configuration via environment variables or BuildConfig
+  private val expectedTenantId: String = System.getenv("MICROSOFT_TENANT_ID") ?: ""
```
Comment (babbahy) [inline app/src/main/java/com/android/sample/auth/MicrosoftAuth.kt]: Your 2 getenv can produce empty strings, which disables tenant enforcement.

File: app/src/main/java/com/android/sample/auth/MicrosoftAuth.kt [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+      val padded =
+          when (payloadB64.length % 4) {
+            2 -> "$payloadB64=="
+            3 -> "$payloadB64="
+            else -> payloadB64
+          }
+
+      val bytes =
+          try {
+            Base64.decode(padded, Base64.DEFAULT)
```
Comment (babbahy) [inline app/src/main/java/com/android/sample/auth/MicrosoftAuth.kt]: what about edge cases ? for example if it wraps over it ?

File: app/src/main/java/com/android/sample/auth/MicrosoftAuth.kt [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+      val auth = Firebase.auth
+
+      // Handle pending auth result (e.g., after rotation)
+      val pending = auth.pendingAuthResult
+      if (pending != null) {
+        Log.d(TAG, "Resuming pending authentication flow")
+        pending
+            .addOnSuccessListener { result ->
+              Log.d(TAG, "Pending authentication successful")
+              handleAuthResult(result.credential as? OAuthCredential, onSuccess, onError)
```
Comment (babbahy) [inline app/src/main/java/com/android/sample/auth/MicrosoftAuth.kt]: when error, it is better it is better to give explicit null check or the clear error.
